community/cloud-foundation/templates/cloud_nat/cloud_nat.py

```diff
@@ -0,0 +1,98 @@
+""" This template creates a Cloud NAT. """
+MIN_PORTS_PER_VM = 64
+UDP_IDLE_TIMEOUT_SEC = 30
+ICMP_IDLE_TIMEOUT_SEC = 30
+TCP_ESTABLISHED_IDLE_TIMEOUT_SEC = 1200
+TCP_TRANSITORY_IDLE_TIMEOUT_SEC = 30
+
+def get_nat_ip_option_enum(natips):
+    """ Returns one of the two supported Enum for Nat IP Option
+        AUTO_ONLY or MANUAL_ONLY """
+    if natips:
+        return 'MANUAL_ONLY'
+    return 'AUTO_ONLY'
+
+def generate_config(context):
+    """ Entry point for the deployment resources. """
+
+    name = context.properties.get('name', context.env['name'])
+
+    resources = [
+        {
+            'name': context.env['name'],
+            # compute.v1.router seems to be doing the job though
```

Comment:
[https://cloud.google.com/compute/docs/reference/rest/v1/routers](https://cloud.google.com/compute/docs/reference/rest/v1/routers) Managed NAT got GA, so itś part of compute.v1